src/sentry/search/snuba/executors.py

```diff
@@ -1374,13 +1510,28 @@ def query(
             # is invalid.
             return self.empty_result
 
+        if len(projects) == 0:
+            return self.empty_result
+
+        organization = projects[0].organization
+
         event_entity = self.entities["event"]
         attr_entity = self.entities["attrs"]
         search_issues_entity = self.entities["search_issues"]
 
         queries = []
-        entities_to_check = [event_entity, search_issues_entity]
+        entities_to_check = []
+        group_categories = group_categories_from_search_filters(search_filters, organization, actor)
+
+        if GroupCategory.ERROR.value in group_categories:
+            entities_to_check.append(event_entity)
+
+        # if we are checking more than one category or we are looking at something other than errors then we need to check search issues
+        if len(group_categories) > 1 or GroupCategory.ERROR.value not in group_categories:
```

Comment:
why do you need to check the length? shouldn't the second condition here be enough?

Reply:
@JoshFerge If we want to search say errors and replays, then we have to search the search issues dataset as well. If we have at least 2 categories, it's guaranteed one of them is non-error which is why we have that OR condition. If I removed `len(group_categories) > 1` then we'd only search search issues if we don't search errors as well.

Reply:
why wouldn't
`any([GroupCategory.ERROR.value != gc for gc in group_categories])` work then?

Reply:
@JoshFerge yep, that would work too. i'll change it.